crates/sui-core/src/execution_cache/writeback_cache.rs

```diff
@@ -1075,14 +1084,28 @@ impl WritebackCache {
     }
 
     // Updates the latest object id cache with an entry that was read from the db.
-    fn cache_latest_object_by_id(&self, object_id: &ObjectID, object: LatestObjectCacheEntry) {
+    fn cache_latest_object_by_id(
+        &self,
+        object_id: &ObjectID,
+        object: LatestObjectCacheEntry,
+        ticket: Ticket,
+    ) {
         trace!("caching object by id: {:?} {:?}", object_id, object);
-        self.metrics.record_cache_write("object_by_id");
-        self.cached.object_by_id_cache.insert(object_id, object);
+        if self
+            .cached
+            .object_by_id_cache
+            .insert(object_id, object, ticket)
+            .is_ok()
+        {
+            self.metrics.record_cache_write("object_by_id");
```

Comment:
nit: might be worthwhile to track the cache writes by ticket type

Reply:
that probably would be useful but i think i'll leave it for a separate PR - the metrics for WB cache are already a bit unwieldy (my fault)